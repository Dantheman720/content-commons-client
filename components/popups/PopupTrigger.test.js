import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import PopupTrigger from './PopupTrigger';

const props = {
  toolTip: 'Share Project',
  icon: { img: 'shareIcon.svg', dim: 18 },
  content: <div>content</div>,
  show: true
};

const Component = <PopupTrigger { ...props } />;

const resizeWindow = ( x, y ) => {
  window.innerWidth = x;
  window.innerHeight = y;
  window.dispatchEvent( new Event( 'resize' ) );
};

describe( '<PopupTrigger />', () => {
  it( 'renders without crashing', () => {
    const wrapper = mount( Component );

    expect( wrapper.exists() ).toEqual( true );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it( 'sets initial state', () => {
    const wrapper = mount( Component );
    const inst = wrapper.instance();

    expect( inst.state ).toEqual( { isMobile: false } );
  } );

  it( 'componentDidMount calls isMobile', () => {
    const wrapper = mount( Component );
    const inst = wrapper.instance();
    const spy = jest.spyOn( inst, 'isMobile' );

    inst.componentDidMount();
    expect( spy ).toHaveBeenCalled();
  } );

  it( 'isMobile sets isMobile to true in state if window.outerWidth is < 600', () => {
    const wrapper = mount( Component );
    const inst = wrapper.instance();

    // initial state
    expect( inst.state.isMobile ).toEqual( false );

    window.outerWidth = 500;
    inst.isMobile();
    expect( inst.state.isMobile ).toEqual( true );
  } );

  it( 'isMobile does not set isMobile to true in state if window.outerWidth is > 600', () => {
    const wrapper = mount( Component );
    const inst = wrapper.instance();

    // initial state
    wrapper.setState( { isMobile: false } );
    expect( inst.state.isMobile ).toEqual( false );

    window.outerWidth = 900;
    inst.isMobile();
    expect( inst.state.isMobile ).toEqual( false );
  } );

  it( 'sets inline display value to none if !props.show', () => {
    const wrapper = mount( Component );
    wrapper.setProps( { show: false } );
    const span = wrapper.find( 'span' );

    expect( span.prop( 'style' ) ).toEqual( { display: 'none' } );
  } );

  it( 'sets correct Popup position and className props', () => {
    const wrapper = mount( Component );
    const popup = () => wrapper.find( 'Popup' );

    expect( wrapper.state( 'isMobile' ) ).toEqual( false );
    expect( popup().prop( 'className' ) ).toEqual( 'popupElem_wrapper' );
    expect( popup().prop( 'position' ) ).toEqual( 'bottom right' );

    wrapper.setState( { isMobile: true } );
    expect( wrapper.state( 'isMobile' ) ).toEqual( true );
    expect( popup().prop( 'className' ) )
      .toEqual( 'popupElem_wrapper popupElem_wrapper--mobile' );
    expect( popup().prop( 'position' ) ).toEqual( 'bottom center' );
  } );
} );
