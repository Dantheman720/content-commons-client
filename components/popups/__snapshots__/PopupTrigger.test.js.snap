// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopupTrigger /> renders without crashing 1`] = `
<PopupTrigger
  content={
    <div>
      content
    </div>
  }
  icon={
    Object {
      "dim": 18,
      "img": "shareIcon.svg",
    }
  }
  show={true}
  toolTip="Share Project"
>
  <span
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <Popup
      className="popupElem_wrapper"
      content={
        <div>
          content
        </div>
      }
      horizontalOffset={4}
      keepInViewPort={true}
      on="click"
      position="bottom right"
      trigger={
        <Button
          as="button"
          className="trigger"
          tooltip="Share Project"
        >
          <img
            alt="Share Project"
            height={18}
            src="shareIcon.svg"
            width={18}
          />
           
        </Button>
      }
    >
      <Portal
        closeOnDocumentClick={true}
        closeOnEscape={true}
        closeOnTriggerClick={true}
        eventPool="default"
        onClose={[Function]}
        onMount={[Function]}
        onOpen={[Function]}
        onUnmount={[Function]}
        openOnTriggerClick={true}
        trigger={
          <Button
            as="button"
            className="trigger"
            tooltip="Share Project"
          >
            <img
              alt="Share Project"
              height={18}
              src="shareIcon.svg"
              width={18}
            />
             
          </Button>
        }
        triggerRef={[Function]}
      >
        <Ref
          innerRef={[Function]}
        >
          <Button
            as="button"
            className="trigger"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            tooltip="Share Project"
          >
            <button
              className="ui button trigger"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              tooltip="Share Project"
            >
              <img
                alt="Share Project"
                height={18}
                src="shareIcon.svg"
                width={18}
              />
               
            </button>
          </Button>
        </Ref>
      </Portal>
    </Popup>
  </span>
</PopupTrigger>
`;
